### PR TITLE
fix(server): inconsistent key set for Garden instances for concurrent requests

### PR DESCRIPTION
**Pull request by edvald**
cc @eysi09, can you take a look to see if this partially addresses #4597?

### CODE REVIEW

**Comment by eysi09** [comment]
I'm still seeing the same behaviour. 

[Here](https://github.com/garden-io/garden/blob/master/core/src/server/instance-manager.ts#L131) the key `env=dev` (for my example project). 

There will never be an instance with this key because when we add it to the map it'll have the key `env=dev-my-namespace`. 

So we always hit the codepath that creates a new Garden instance. That code path also unsubscribes the listener from the previous instance so we drop any further events from that instance. 

Basically what's described here: #4597.

Note also that `defaultNs` resolves to `${var.dev-env-name}` on my project (see my comment on the code).

File: core/src/garden.ts [hunk 1]
```diff
@@ -1506,11 +1506,7 @@ export class Garden {
   public getInstanceKeyParams() {
     let namespace: string | undefined
 
-    let defaultNs = this.getEnvironmentConfig().defaultNamespace
-
-    if (defaultNs === undefined) {
-      defaultNs = defaultNamespace
-    }
+    const defaultNs = this.getEnvironmentConfig().defaultNamespace || defaultNamespace
```
Comment (eysi09) [inline core/src/garden.ts:1509]: The template string isn't resolved here so I get `defaultNs=${var.dev-env-name}`.

The returned namespace is still `vote-demo-eysi` because of the line below:

``` 
    if (this.namespace !== defaultNs) {
      namespace = this.namespace
    }
```

**Review by eysi09 — CHANGES_REQUESTED** [review]

**Comment by eysi09** [comment]
...what I don't really understand is why we first read the key from the instance manager params and then again in the `set` from the Garden instance. 

I would've assumed we'd just re-use the original key we get from params and use that in the `set` method.

**Comment by eysi09** [comment]
Some further findings:

When we create the initial DummyGarden for the parent dev command the `gardenKey` we [set on the log context](https://github.com/garden-io/garden/blob/master/core/src/garden.ts#L1855) has a namespace (default) so it becomes `env=default-default`. 

I think that conflicts with this code comment: `// Setting this after resolving the gardenKey above because we don't want the default namespace resolved there`

When we then create a "proper" Garden instance via the instance manager the namespace is missing so the log context gets `gardenKey=env=default`.

**Comment by edvald** [comment]
Closed in favor of #4649
